apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/PaginationControl/index.tsx

```diff
@@ -46,24 +41,31 @@ export const PaginationControls = (props: PaginationControlProps) => {
       <PageLink onClick={() => setCurrentPage(totalPages)} disabled={currentPage === totalPages}>
         Last
       </PageLink>
-      <PageLink>
-        <Select 
-          value={dropdownOptions.filter((v) => v.value === rowsPerPage)[0].value} 
-          onChange={(e) => setRowsPerPage(Number(e.currentTarget.value))}>
-          {dropdownOptions.map(o=>(
-            <option key={o.value} value={o.value}>{o.label}</option>
-          ))}
-        </Select>
-      </PageLink>
+      <InputSelect
+        onChange={(e:Option) => setRowsPerPage(Number(e.value))}
+        options={dropdownOptions}
+        defaultValue={dropdownOptions.find(o => o.value === rowsPerPage)}
+        width='6rem'
+        fontSize='0.68rem'
+        zIndex={5} // place it above table filter dropdowns when they overlap on mobile
+      />
     </PaginationContainer>
   );
 };
 
+const PaginationContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  text-align: right;
+  margin: 10px 0;
+`;
+
 type PageLinkProps = {
   selected?: boolean;
+  disabled?: boolean;
 };
 
-const PageLink = styled.button<PageLinkProps>`
+export const PageLink = styled.button<PageLinkProps>`
```

Comment:
Minor thing: It would be nice to change to a darker color maybe if the link is disabled (in addition to the 'not-allowed' cursor). But this is non-blocking. Not a big issue.